### DISCUSSION TITLE
Release planning v8.1

### DISCUSSION THREAD

**Discussion by dev-zero** [cp2k/cp2k#2600] [Team Posts]
Dear all,

since we should be doing another release I've created a milestone on GitHub and assigned issues and PRs which would make good targets for release v8.1, ideally scheduled for December 2020:

  https://github.com/cp2k/cp2k/milestone/4

For the library targets (libxc-5, libint-2.7) we depend on upstream to make new releases, but those targets can be dropped if needed and introduced later in a v8.2 release. [DBCSRs roadmap](https://github.com/cp2k/dbcsr/milestone/5) for its final v2.1 release will also be streamlined a bit to be able to make it. Other goals on the milestone can of course also be a topic of discussion.

Please take a look at the list of issues, give feedback where needed or even better, take responsibility for one or more issues!
Also, please spread the word to submit PRs if you know developers sitting on unsubmitted changes!

Best regards,
Tiziano

**Comment by oschuett** [comment]
The end of 2020 is drawing close. I believe all blocking issues have been [resolved](https://github.com/cp2k/cp2k/milestone/4) and my GPU [work](https://github.com/cp2k/cp2k/pull/1237) should go in today. 

So, shall we aim for another Christmas release?

**Comment by dev-zero** [comment]
Unfortunately not: at least for libvori we have another upgrade to be done, see https://github.com/cp2k/cp2k/issues/1229

**Comment by dev-zero** [comment]
and this PR also needs to be merged: https://github.com/cp2k/cp2k/pull/1195

**Comment by dev-zero** [comment]
it would also be great if someone could go over the packages in the toolchain again and check whether we have some minor package upgrades which could be pulled in before the relase

**Comment by dev-zero** [comment]
... and there's also the question about libxc 5.1: the devs seem to be at work to push out another release soon, with the Fortran API changes reverted. Should we wait and see or do we go with libxc 4.x for the CP2K v8.1 release?

**Comment by dev-zero** [comment]
furthermore #814 seems like an easy win now with DBCSR v2.1, or not?

**Comment by alazzaro** [comment]
Well, you need to introduce the set_device for the PW_CUDA, so that you don't need DBCSR at all.

**Comment by mkrack** [comment]
A release by December 24 seems to me a quite ambitious project. We have had normally a grace time of about two weeks for testing and validation without major commits before a release.

**Comment by oschuett** [comment]
I really would like to have a release this year.

The last major change was to the [active space module](https://github.com/cp2k/cp2k/commit/6964239efd08d4cc292ef34491395385576eab63) three days ago. So, then let's aim for December 30th and use the time for testing and polishing.

Obviously, my [GPU kernel](https://github.com/cp2k/cp2k/pull/1237) arrives a bit late, but thanks to the validation against the reference backend I consider it rather low risk. Nevertheless, I plan to mark it as _experimental_ in the release notes.

@dev-zero, I think all those issues are valid and nice to have, but they should wait for the next release.

**Comment by oschuett** [comment]
I've gone over this year's commits and gathered more entries for our release notes.
You can check the draft at: https://www.cp2k.org/version_history

**Comment by martinbrehm** [comment]
Hi @oschuett ,

I recommend to change the libvori entry to the following two points:

(-) Add support for Voronoi integration of electron density ( https://brehm-research.de/voronoi )
(-) Add support for output of electron density in compressed BQB format ( https://brehm-research.de/bqb )

Thanks,
Martin

**Comment by oschuett** [comment]
I think, we are ready to tag the release today.

The dashboard still shows a few failures, but none of them should block us:
- i386: https://github.com/cp2k/cp2k/issues/1118
- GCC 6: https://github.com/cp2k/cp2k/issues/1203
- Farming: https://github.com/cp2k/cp2k/issues/1211
- Cuda: https://github.com/cp2k/cp2k/issues/1243
